src/com/softwaret/kpdf/service/publication/PublicationsServiceImpl.kt

```diff
@@ -24,4 +24,18 @@ class PublicationsServiceImpl : PublicationsService {
                 pdf = Pdf.new { blobPdf = SerialBlob(pdfBase64.value.toByteArray()) }
             }
         }.id.value.asId()
+
+    override fun deletePublication(id: Id) {
+        transaction {
+            Publication.findById(id.value)?.delete() ?: error("Publication to be deleted not found!")
```

Comment:
Sprawdzałeś co się stanie gdy `?: error("....")` rzuci nam dodatkowo wyjątkiem, wypadałoby go złapać i obsłużyć. Chyba że sprawdzałeś, że nie poleci żadna `500` od serwera?

Reply:
Dostaniemy 500 w ryj i bardzo dobrze, bo powinniśmy na wyższych poziomach upewnić się że usuwana publikacja istnieje. Tak samo w przypadku update

Reply:
To łapanka i przemapowanie na 404, żeby klient mógł jakoś obsłużyć